packages/core/src/document.ts

```diff
@@ -584,56 +586,57 @@ class Document extends EventEmitter implements DocStateHolder {
   /**
    * Request anchor for the latest document state
    */
-  async anchor (): Promise<void> {
+  anchor(): void {
     // eslint-disable-next-line @typescript-eslint/no-this-alias
     const doc = this
     const requestTip: CID = this.tip
 
-    this._context.anchorService.on(this.id.toString(), async function listener (asr: AnchorServiceResponse) {
-      if (!asr.cid.equals(requestTip)) {
-        // This message is about a different tip for the same document
-        return
-      }
-      switch (asr.status) {
-        case 'PENDING': {
-          const state = doc._doctype.state
-          state.anchorScheduledFor = asr.anchorScheduledFor
-          doc._doctype.state = state
-          await doc._updateStateIfPinned()
-          return
-        }
-        case 'PROCESSING': {
-          const state = doc._doctype.state
-          state.anchorStatus = AnchorStatus.PROCESSING
-          doc._doctype.state = state
-          await doc._updateStateIfPinned()
-          return
-        }
-        case 'COMPLETED': {
-          doc._context.anchorService.removeListener(doc.id.toString(), listener)
-
-          await doc._handleTip(asr.anchorRecord)
-          await doc._updateStateIfPinned()
-          await doc._publishTip()
-          return
-        }
-        case 'FAILED': {
-          doc._context.anchorService.removeListener(doc.id.toString(), listener)
-          if (requestTip !== doc.tip) {
-            return
-          }
-
-          const state = doc._doctype.state
-          state.anchorStatus = AnchorStatus.FAILED
-          doc._doctype.state = state
-          return
-        }
-      }
-    })
-    await this._context.anchorService.requestAnchor(this.id.toString(), this.tip)
-    const state = this._doctype.state
-    state.anchorStatus = AnchorStatus.PENDING
-    this._doctype.state = state
+    const anchorStatus$ = this._context.anchorService.requestAnchor(this.id.baseID, this.tip);
+    const subscription = anchorStatus$
+        .pipe(
+            concatMap(async (asr) => {
+              if (!asr.cid.equals(requestTip)) {
```

Comment:
Can we get rid of this now that we're getting an observable per anchor request instead of a single event feed from the entire anchor service?  The Observable returned from `AnchorService.requestAnchor` should be able to only emit events that are responses to the specific anchor request right?

Reply:
True.